pkg/api/api.go

```diff
@@ -63,12 +64,15 @@ func prefixError(name string, reqs []Request) []Request {
 		rq := r
 
 		res = append(res, func(ctx context.Context, ch chan<- *Record) error {
-			logger.Info(fmt.Sprintf("Snapshot: starting %s", name))
+			logger := logr.FromContextOrDiscard(ctx)
+			logger.V(6).Info("snapshot start", "action", name)
 
 			if err := rq(ctx, ch); err != nil {
 				return fmt.Errorf("failed to run %q: %w", name, err)
 			}
 
+			logger.V(6).Info("snapshot complete", "action", name)
```

Comment:
Best to defer this at the beginning of the function in order to capture the case where the function exits with an error. You may want to include `error`.

Reply:
The error gets returned, so I'm not as worried about logging it here. Also since golang doesn't have ternary statements, there's no clean way to write the defer as a single line that can indicate either success or failure. The specific complaint I'm trying to address with this logging is that it's hard to see when snapshot is running normally/successfully and what it's been doing. The errors are correctly propagating out